### PR TITLE
TDT-155 Implement subqueries

### PR DESCRIPTION
**Pull request by alexmk92**

### CODE REVIEW

File: lib/Doctrine/Lib.php [hunk 1]
```diff
@@ -268,6 +268,36 @@ public static function arrayDeepMerge()
...
+            }
+
+            if (is_array($value) && is_array($compare[$key])) {
+                $destination[$key] = array_merge($destination[$key], $compare[$key]);
+            } else if (!empty($compare[$key])) {
+                self::arrayDeepCombine($destination[$key], $compare[$key]);
+            }
+        }
+
+        return $source;
```
Comment (alexmk92) [inline lib/Doctrine/Lib.php]:
```suggestion
        return $destination;
```

File: lib/Doctrine/Query.php [hunk 2]
```diff
@@ -1035,10 +1035,11 @@ protected function _buildSqlFromPart($ignorePending = false)
             if ($k === 0) {
                 if ( ! $ignorePending && $this->_type == self::SELECT) {
                     // We may still have pending conditions
+                    $alias_lookup = empty($this->_derivedTable) ? $e[1] : end($e);
                     $alias = count($e) > 1
                         ? $this->getComponentAlias($e[1])
                         : null;
-                    $where = $this->_processPendingJoinConditions($alias);
+                    $where = $this->_processPendingJoinConditions($alias_lookup);
```
Comment (alexmk92) [inline lib/Doctrine/Query.php]:
```suggestion
                    $where = $this->_processPendingJoinConditions($alias);
```

File: lib/Doctrine/Query.php [hunk 3]
```diff
@@ -1035,10 +1035,11 @@ protected function _buildSqlFromPart($ignorePending = false)
             if ($k === 0) {
                 if ( ! $ignorePending && $this->_type == self::SELECT) {
                     // We may still have pending conditions
+                    $alias_lookup = empty($this->_derivedTable) ? $e[1] : end($e);
                     $alias = count($e) > 1
                         ? $this->getComponentAlias($e[1])
```
Comment (alexmk92) [inline lib/Doctrine/Query.php]:
```suggestion
                        ? $this->getComponentAlias($alias_lookup)
```

**Review by akhumphrey — APPROVED** [review]
Really lovely work mate 👍🏻

**Pull request merged**